x/ecocredit/server/core/query_balances.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	"context"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types/ormutil"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) Balances(ctx context.Context, req *core.QueryBalancesRequest) (*core.QueryBalancesResponse, error) {
+	acc, err := sdk.AccAddressFromBech32(req.Account)
+	if err != nil {
+		return nil, err
+	}
+
+	pg, err := ormutil.GogoPageReqToPulsarPageReq(req.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.BatchBalanceTable().List(ctx, api.BatchBalanceAddressBatchIdIndexKey{}.WithAddress(acc), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
+
+	balances := make([]*core.BatchBalance, 0, 10)
```

Comment:
why starting with cap = 10?

Reply:
i was thinking it might be good to add some arbitrary cap to avoid slice allocations. would you advise a smaller/bigger number or just not at all? maybe we could add a `default` constant if we want to define a capacity?

Reply:
No idea what would be the most common one, that's why I asked ;) Maybe let's decrease to 8 and add a comment?
By default, go will double the capacity if it needs.